bridge/bridge.go

```diff
@@ -20,12 +20,13 @@ type Config struct {
 	// Map from Discord to IRC
 	ChannelMappings map[string]string
 
-	IRCServer        string
-	IRCServerPass    string
-	IRCListenerName  string // i.e, "DiscordBot", required to listen for messages in all cases
-	WebIRCPass       string
-	NickServIdentify string // string: "[account] password"
-	PuppetUsername   string // Username to connect to IRC with
+	IRCServer         string
+	IRCServerPass     string
+	IRCListenerName   string          // i.e, "DiscordBot", required to listen for messages in all cases
+	IRCIgnoresDiscord map[string]bool // Ignore Discord nicks on IRC side of the relay
```

Comment:
this could be a `map[string]struct{}` instead

Reply:
Why would a map[string]struct{} be appropriate? I'm not claiming to be a golang master or anything but are we looking to pack values into the map other than "should ignore/shouldn't ignore"? The idea was if one is say developing a bot it would mean they might for some reason have it in a spammy mode but would wish to not allow the relay to send it into IRC, but would just set it back to false later.

Reply:
this kinda links into the other thread about it being a `[]string`. if it's a `[]string`, we only need to test for presence in the map, instead of checking the value. (and, less importantly,`struct{}` occupies no space in memory)

```go
m := make(map[string]struct{})
m["someone"] = struct{}{} // add to map
_, ok := m["someone"] // test for map presence, map value ignored
delete(m, "someone") // delete from map
```

Reply:
oh and about "setting it to false", they could always comment out the line

Reply:
You're correct as to commenting it out. I'll update this to handle using `[]string` and `map[string]struct{}` soon.

Reply:
This should be done now, note I haven't tested it yet. The code should be using `map[string]struct{}` but reading in a `[]string` from the config.